lib/ar_state_machine.rb

```diff
@@ -352,7 +352,13 @@ def process_callbacks(to, model, from, callbacks, ignore_result = false)
             break # just exit THIS chain; not the whole rails callback chain
           else
             # rollback changes / cancel subsequent other callbacks
-            throw :abort
+            if raise_not_throw
+              # after save callback chain is halted like this
+              # https://github.com/rails/rails/issues/33192
+              raise ActiveRecord::RecordInvalid, model
```

Comment:
I assume we'll still have access to whatever error is on the model?

Reply:
yep yep.